### DISCUSSION TITLE
I want web components to enable exposing elements to assistive technology and forms

### DISCUSSION THREAD

**Discussion by aarongustafson** [WebWeWant/webwewant.fyi#152] [Wants]
Currently when using web components and `shadowRoot`, any element within the `shadowRoot` will not be noticed by, say a screen readers, or even a form (for example when doing a form `input`). There is an effort to come up with a solution to the form part, but that would not automatically solve the assistive technology part. It would be awesome to be able to add some ARIA/`role` attributes to the component tag or identify elements that should be exposed within the `shadowRoot` to make the the component accessible.

<hr>

https://webwewant.fyi/wants/86/

**Comment by carmacleod** [comment]
This might be helpful: https://wicg.github.io/aom/explainer.html
